app/models/user.rb

```diff
@@ -3,4 +3,22 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+  serialize :availibility_start_time, Tod::TimeOfDay
+  serialize :availibility_end_time, Tod::TimeOfDay
+
+  validates_presence_of :nickname, :first_name, :last_name
+  validates_inclusion_of :actor, in: [true, false]
+
+  validate :start_time_cannot_be_greater_or_equal_end_time, if: :actor?
+
+  with_options if: :actor? do
+    validates :availibility_start_time, presence: true
+    validates :availibility_end_time, presence: true
+  end
+
+  def start_time_cannot_be_greater_or_equal_end_time
+    if availibility_start_time > availibility_end_time || availibility_start_time == availibility_end_time
+      errors.add(:availibility_start_time, "can't be later or equal to end time")
```

Comment:
modify
errors.add(:availability_start_time, "must be earlier to availability end time")
errors.add(:availability_end_time, "must be later to availability end time")